vignettes/advs.Rmd

```diff
@@ -0,0 +1,407 @@
+the SDTM `--TESTCD` value to join to the source data.
+The only difference here would be the the additional parameters
+collected in metabolic trials - `BMI`, `HIPCIR` and `WSTCIR`.
+
+```{r, echo=TRUE, message=FALSE}
+param_lookup <- tribble(
+  ~VSTESTCD, ~PARAMCD, ~PARAM, ~PARAMN, ~PARCAT1, ~PARCAT1N,
+  "HEIGHT", "HEIGHT", "Height (cm)", 1, "Subject Characteristic", 1,
+  "WEIGHT", "WEIGHT", "Weight (kg)", 2, "Subject Characteristic", 1,
+  "BMI", "BMI", "Body Mass Index(kg/m^2)", 3, "Subject Characteristic", 1,
+  "HIPCIR", "HIPCIR", "Hip Circumference (cm)", 4, "Subject Characteristic", 1,
+  "WSTCIR", "WSTCIR", "Waist Circumference (cm)", 5, "Subject Characteristic", 1,
+  "DIABP", "DIABP", "Diastolic Blood Pressure (mmHg)", 6, "Vital Sign", 2,
+  "PULSE", "PULSE", "Pulse Rate (beats/min)", 7, "Vital Sign", 2,
+  "SYSBP", "SYSBP", "Systolic Blood Pressure (mmHg)", 8, "Vital Sign", 2,
+  "TEMP", "TEMP", "Temperature (C)", 9, "Vital Sign", 2
+)
+
+attr(param_lookup$VSTESTCD, "label") <- "Vital Signs Test Short Name"
+```
+
+This lookup may now be joined to the source data and this is how the
+parameters will look like:
+
+```{r, eval=TRUE, include=TRUE, message=FALSE}
+advs <- derive_vars_merged_lookup(
+  advs,
+  dataset_add = param_lookup,
+  new_vars = exprs(PARAMCD, PARAM, PARCAT1, PARCAT1N),
+  by_vars = exprs(VSTESTCD)
+)
+```
+
+```{r, eval=TRUE, echo=FALSE}
+advs_param <- distinct(advs, USUBJID, PARAMCD, VSTESTCD, PARAM, PARCAT1, PARCAT1N)
+
+dataset_vignette(advs_param, display_vars = exprs(USUBJID, VSTESTCD, PARAMCD, PARAM, PARCAT1, PARCAT1N))
+```
+
+The mapping of `AVAL` is left to the ADaM programmer. An example mapping may be:
+
+```{r, eval=TRUE}
+advs <- mutate(
+  advs,
+  AVAL = VSSTRESN
+)
+```
+
+## Derive Additional Parameters (e.g. `BMI` for `ADVS`) {#derive_param}
+
+In metabolic trials, `BMI` is often calculated at source using collected
+height and weight for screening purposes. But while creating the `ADVS`
+dataset, we re-derive `BMI` from the collected height and weight values.
+This is done to ensure that the `BMI` is calculated consistently across
+all subjects and visits.
+
+In this step, we create parameter Body Mass Index (`BMI`) for the `ADVS`
+domain using the `derive_param_bmi()` function. Note that only variables
+specified in the `by_vars` argument will be populated in the newly
+created records. Also note that if height is collected only once for a
+subject use `constant_by_vars` to specify the function to merge by the
+subject-level variable - otherwise BMI is only calculated for visits
+where both are collected.
+
+```{r eval=TRUE}
+# Removing BMI collected at source from the dataset
+advs <- advs %>% filter(!VSTESTCD == "BMI")
+
+advs <- derive_param_bmi(
+  advs,
+  by_vars = c(
+    get_admiral_option("subject_keys"),
+    exprs(!!!adsl_vars, VISIT, VISITNUM, ADT, ADY, VSTPT, VSTPTNUM)
+  ),
+  set_values_to = exprs(
+    PARAMCD = "BMI",
+    PARAM = "Body Mass Index (kg/m^2)",
+    PARAMN = 3,
```

Comment:
It seems that in this step PARAMN for BMI is set, but at the same time PARAMN for all the other parameters is set to NA, i.e. the PARAMN values that were previously there for the other parameters do not remain.

Reply:
Are you sure? This would be a bug in the admiral function. The values specified for `set_values_to` should be used for the new records only.

From the code it seems that `PARAMN` is not derived before. It is included in the lookup table but not merged to the data (`PARAMN` is not included in `new_vars` in the `derive_vars_merged_lookup()` call).

Reply:
You are right. Thanks @bundfussr.
I had seen the `PARAMN` only in the lookup table but not in the ADVS dataset itself. I will just make a suggestion to add `PARAMN` for the other parameters so that not only BMI get's a `PARAMN`.